sound/soc/bcm/iqaudio-dac.c

```diff
@@ -110,6 +170,45 @@ static int snd_rpi_iqaudio_dac_probe(struct platform_device *pdev)
 	    if (of_property_read_string(pdev->dev.of_node, "dai_stream_name",
 					&dai->stream_name))
 		dai->stream_name = "IQaudIO DAC HiFi";
+
+	    /* gpio_unmute - one time unmute amp using GPIO */
+	    gpio_unmute = of_property_read_bool(pdev->dev.of_node,
+						"iqaudio-dac,unmute-amp");
+	    /* auto_gpio_mute - mute/unmute amp using GPIO */
+	    auto_gpio_mute = of_property_read_bool(pdev->dev.of_node,
+						   "iqaudio-dac,auto-mute-amp");
+	    if (auto_gpio_mute || gpio_unmute) {
+		struct device_node *pins_node =
+			of_parse_phandle(pdev->dev.of_node, "pinctrl-0", 0);
+		if (pins_node) {
+		    u32 pin, function;;
+		    if (of_property_read_u32_index(pins_node, "brcm,pins",
+						   0, &pin) == 0)
+			if (of_property_read_u32_index(pins_node,
+					"brcm,function", 0, &function) == 0)
+			    if (function == 1) // out
+				gpio_mute_pin = pin;
+		}
+
+		if (gpio_mute_pin) {
+			gpiochip = gpiochip_find("pinctrl-bcm2835",
```

Comment:
This is no longer the recommended way to use GPIOs; take a look at the ADAU1977 driver and overlay and how the reset line is handled.